packages/cdk8s/lib/app.ts

```diff
@@ -32,12 +37,140 @@ export class App extends Construct {
    * Synthesizes all manifests to the output directory
    */
   public synth(): void {
-    if (!fs.existsSync(this.outdir)) {
-      fs.mkdirSync(this.outdir);
+
+    fs.mkdirSync(this.outdir, { recursive: true });
+
+    // this is kind of sucky, eventually I would like the DependencyGraph
+    // to be able to answer this question.
+    const hasChartDependencies = App.prepare(this);
```

Comment:
Can you please move all these static methods as free-standing functions in a separate (internal) file called `synthesis.ts`

Reply:
Agreed on leaving them as internal module functions in `app.ts`